### ISSUE TITLE
After 2.2 update, Xbox One S controller is no longer working

### ISSUE THREAD

**Issue by billkings95** [OpenEmu/OpenEmu#4073]
## 1 - Summary of issue

Xbox One controller (bluetooth, model S) is no longer working correctly. Analog sticks no longer mapped by the emulator.

## 2 - Steps to reproduce

1. Update OpenEmu 2.1.1 to 2.2;
2. Connect Xbox One S Pad by Bluetooth, using macOS Catalina official controller drivers. Pad with batteries fully recharged;
3. When trying to map the Xbox One controller, the analog L and R is no longer working properly. OpenEmu cannot map the analog sticks anymore or is mapping them in wrong places (e.g. when trying to map Y- axis for UP the emulator automatically sets Y- for LEFT, not allowing for manual correction);
4. Reverting OpenEmu to 2.1.1 solves the problem, when the Pad is fully working.

## 3 - Observed Behavior

1. Controller doesn't work via Bluetooth or wired;
2. Pad is brand new, used without problems in previous builds of OpenEmu;
3. The problem still continues even using third-party drivers (unofficial FranticRain's 360Controller);
4. The problem still continues in previous macOS release (Mojave, 10.14.5);
5. Neither updating or doing a clean OpenEmu installation seems to solve the problem. Only reverting to the previous version.

## 4 - Debugging Information

1. OpenEmu: Version 2.2;
2. Hardware: Mac mini Late 2014 with macOS Catalina 10.15.1.

**Comment by clobber** [comment]
> by Bluetooth, using macOS Catalina official controller drivers

What is that exactly? The “Xbox One S” is Bluetooth and requires no driver for macOS. There is no “official” driver.

I’d uninstall whatever you have, be it 360Controller or whatever, and just pair your controller over Bluetooth like normal and that ought to work fine.

**Comment by billkings95** [comment]
I decided to format my Mac mini and reinstall MacOS Catalina. The problem still happens in version 2.2, connecting the gamepad via Bluetooth. 

The control is only working normally when I revert to version 2.1.1. :(

**Comment by clobber** [comment]
Ping @MaddTheSane can you replicate this?

**Comment by MaddTheSane** [comment]
Can confirm: the Xbox One S Bluetooth controller has issues with OpenEmu 2.2 under Catalina. After inputting the d-pad for a PS1 controller, I got to the left analog input and it started misbehaving, not letting me enter in the left-right part of the left analog, and not letting me enter in any other buttons. Quitting and restarting OpenEmu lets me enter in the buttons.

**Comment by clobber** [comment]
@MaddTheSane thanks for confirming. Could we get an output from HIDBrowser so we can see the HID profile?

**Comment by MaddTheSane** [comment]
![Screen Shot 2019-12-30 at 1 49 12 PM copy](https://user-images.githubusercontent.com/719492/71600367-d9630000-2b0b-11ea-9321-2b8c984bc49f.png)

A thought: OpenEmu uses the GameController framework, right? If it does, could the HID monitoring and the GameController framework cause conflicts?

**Comment by clobber** [comment]
> OpenEmu uses the GameController framework, right?

Nope, not using that framework for gamepad input. Everything comes in through the IOKit level, in OpenEmu-SDK. One of the fixes for other input issues must’ve broken analogs for the Xbox One S. If you revert some commits in OpenEmu-SDK you could pinpoint which one. At the moment, I don’t believe any one else has this gamepad to test with.

**Comment by PsychoH13** [comment]
Can you export the profile of that controller and send it over?

**Comment by clobber** [comment]
@MaddTheSane here is a version of HIDBrowser than can export the HID profile just in case yours cannot. [HIDBrowser.zip](https://github.com/OpenEmu/OpenEmu/files/4011286/HIDBrowser.zip)

**Comment by MaddTheSane** [comment]
Here's the exported profile:
[Xbox Wireless Controller.plist.zip](https://github.com/OpenEmu/OpenEmu/files/4011294/Xbox.Wireless.Controller.plist.zip)

**Comment by clobber** [comment]
@billkings95 We believe the analog stick problem is at least fixed. Please test this build: https://send.firefox.com/download/4f2509ab0285bff0/#pPaiwLm2QpGQpd-LcKoKWg

**Comment by billkings95** [comment]
I can confirm that the analog sticks are now mapped and working correctly. 

However, I have noticed that occasionally, after correctly mapping the analogs, the following buttons (e.g. B, A, Y, X, Start, etc...) cannot be remapped. I press the buttons on the gamepad and nothing happens. 

This behavior happened twice in a sequence of 5 tests (mapping the PlayStation and GameCube controllers). However, when I reopen the emulator, it allows me to set the other buttons. 

Thanks guys for all the help. :)

**Comment by shysaur** [comment]
When the other buttons cannot be mapped after mapping the analog, please try rotating the analog sticks in all directions. Then report back here if doing so successfully "unlocks" the other mappings.

**Comment by clobber** [comment]
@billkings95 Thank you for your diligence in testing. Please thoroughly test this new build to see if that resolves the matter of buttons not mapping:

https://send.firefox.com/download/252626e07db00654/#MQKyyqlBcBZLEXljAqH_pA

**Comment by billkings95** [comment]
@shysaur When testing *OpenEmu_axis-fix_1*, rotating the analog sticks in all directions seems to unlock the other buttons. 

@clobber In *OpenEmu_axis-fix_2*, the problem doesn't happen anymore.

By the way, since I don't have other controllers to test anymore (PS3 and PS4), I can't tell if it's a bug, but in this current version (2.2 and fixes), I also noticed that when mapping the controllers, the buttons seems very sensitive. For example, previously (2.1.1) when mapping an analog or shoulder button/trigger, it was required to press them for a little while (milliseconds), now with any subtle or accidental movement the app understands that I am pressing the buttons to be mapped. I don't know if I was able to express myself correctly, since I'm not so proficient in English when writing. 

Anyway, thanks guys for all the help in solving this bug. :)
